### DISCUSSION TITLE
Upgrading astro-paper version for a site

### DISCUSSION THREAD

**Discussion by emeraldjava** [satnaing/astro-paper#393] [General]
Hi,

I'm curious to know how other user handle the upgrade of the astro-paper theme on their sites.

I've a basic manual workflow which does these steps
1 - create a clean astro site with latest astro-paper theme in a new empty dir.
2 - so a 'diff' between the various files and folders.
3 - merge back the 'latest' changes from astro-paper to my site and commit.

This is fine but assumes a level of confidence and skill with git merging etc.

I guess this is a broader 'astro' issue but it seems to me, that there could be a cleaner and clearer boundary between the 'theme' code and the 'user' files and folders here.  This could be a fairly major change to the astro platform.

I do wonder in the short term, is the concepts of an 'astro-theme-updater' module which is imported via the package.json devDependencies. The version of the 'astro-paper' theme would also be defined in the package.json, and the updater tool would go off, do the latest extract and create the changes on a new branch. It means a site user, could be presented with an auto-mated PR with the layout/config changes they need to commit to get to the latest theme version. I'm just missing the technical steps on how this could be implemented.

Interested in others thoughts.
Thanks,
emeraldjava

See
- https://github.com/emeraldjava/emeraldjava.github.io
- https://emeraldjava.github.io

**Comment by eerison** [comment]
Hey man there is a post about it: https://astro-paper.pages.dev/posts/how-to-update-dependencies/#updating-astropaper-using-git

there are 2 ways, I am using via git, But not as the post ... But in theory it should work fine :D

what I am doing is basically, revert all commit from astro-paper in one, and add as my first commit.
if you are using merge instead of rebase , I guess it will be a pain.

it is my repo: https://github.com/shield-wall/erison-work/commits/master/

in case you find a simple way to handle it, please share here :)

**Reply by emeraldjava** [reply]
Thanks @eerison  - nice write up on the git merge approach. I'll borrow some of your steps. Well done on your site.

**Comment by k4u5hik** [comment]
Many Thanks.

**Comment by emeraldjava** [comment]
I have opened this discussion topic on the main astrology forum to request comment.

https://github.com/withastro/roadmap/discussions/959

**Comment by eerison** [comment]
Hello guys :)

just to share with you that I created 2 articles with alternatives how to update your blog.

an simple one: https://blog.erison.work/posts/get-frontend-theme-updates-using-git-part-1/
and another one keeping the git history clean: https://blog.erison.work/posts/get-frontend-theme-updates-using-git-part-2/

I split into 2 articles, because it gonna be so big.

I hope it helps you update your projects, and in case you face any issue, let me know in the comments :)

**Comment by popovicu** [comment]
I first created my website 2 years ago with this and I'm trying to do an update now. I know, it's long overdue, but I'm unable to update. It's a super time consuming process and I ended up just reverting anyway.

Mixing the project code with the user content and expecting the users to work through the diffs is a very suboptimal experience, unfortunately.
